.github/workflows/rust.yml

```diff
@@ -41,8 +41,8 @@ jobs:
       - name: Build
         run: cargo build --all-features --all-targets --verbose
 
-      - name: Run all tests except doctests
-        run: cargo test --all-features --all-targets --verbose
+      - name: Run all unit + library tests (not doc or integration tests)
+        run: cargo test --all-features --all-targets --lib --verbose 
```

Comment:
This seems to not exclude the integration tests, but I guess you figured given your last comment on this PR.

Reply:
Fixed this, see note below. I needed to remove `--all-targets` and replace it with `--lib --bins`, cargo was choosing the most expansive option.